registry/txt.go

```diff
@@ -0,0 +1,183 @@
+
+	ownerMap := map[string]string{}
+
+	for _, record := range records {
+		if record.RecordType != "TXT" {
+			endpoints = append(endpoints, record)
+			continue
+		}
+		ownerID := im.extractOwnerID(record.Target)
+		if ownerID == "" {
+			//case when value of txt record cannot be identified
+			//record will not be removed as it will have empty owner
+			endpoints = append(endpoints, record)
+			continue
+		}
+		endpointDNSName := im.mapper.toEndpointName(record.DNSName)
+		ownerMap[endpointDNSName] = ownerID
+	}
+
+	for _, ep := range endpoints {
+		ep.Labels[endpoint.OwnerLabelKey] = ownerMap[ep.DNSName]
+	}
+
+	return endpoints, err
+}
+
+// ApplyChanges updates dns provider with the changes
+// for each created/deleted record it will also take into account TXT records for creation/deletion
+func (im *TXTRegistry) ApplyChanges(zone string, changes *plan.Changes) error {
+	filteredChanges := &plan.Changes{
+		Create:    changes.Create,
+		UpdateNew: filterOwnedRecords(im.ownerID, changes.UpdateNew),
+		UpdateOld: filterOwnedRecords(im.ownerID, changes.UpdateOld),
+		Delete:    filterOwnedRecords(im.ownerID, changes.Delete),
+	}
+
+	for _, r := range filteredChanges.Create {
+		txt := endpoint.NewEndpoint(im.mapper.toTXTName(r.DNSName), im.getTXTLabel(), "TXT")
+		filteredChanges.Create = append(filteredChanges.Create, txt)
+	}
+	for _, r := range filteredChanges.Delete {
+		txt := endpoint.NewEndpoint(im.mapper.toTXTName(r.DNSName), im.getTXTLabel(), "TXT")
+		filteredChanges.Delete = append(filteredChanges.Delete, txt)
+	}
+
+	return im.provider.ApplyChanges(zone, filteredChanges)
+}
+
+/**
+  TXT registry specific private methods
+*/
+
+func (im *TXTRegistry) getTXTLabel() string {
+	return fmt.Sprintf(txtLabelFormat, im.ownerID)
+}
+
+func (im *TXTRegistry) extractOwnerID(txtLabel string) string {
+	if matches := txtLabelRegex.FindStringSubmatch(txtLabel); len(matches) == 2 {
+		return matches[1]
+	}
+	return ""
+}
+
+/**
+  nameMapper defines interface which maps the dns name defined for the source
+  to the dns name which TXT record will be created with
+*/
+
+type nameMapper interface {
+	toEndpointName(string) string
+	toTXTName(string) string
+}
+
+type prefixNameMapper struct {
+	prefix string
+}
+
+var _ nameMapper = &prefixNameMapper{}
+
+func newPrefixNameMapper(prefix string) *prefixNameMapper {
+	prefix = strings.TrimSuffix(prefix, ".") + "."
+	return &prefixNameMapper{prefix: prefix}
+}
+
+func (pr *prefixNameMapper) toEndpointName(txtDNSName string) string {
+	if strings.HasPrefix(txtDNSName, pr.prefix) {
+		return strings.TrimPrefix(txtDNSName, pr.prefix)
+	}
+	return ""
+}
+
+func (pr *prefixNameMapper) toTXTName(endpointDNSName string) string {
+	return pr.prefix + endpointDNSName
+}
+
+type noopNameMapper struct {
+}
+
+var _ nameMapper = &noopNameMapper{}
+
+func newNoopNameMapper() *noopNameMapper {
+	return &noopNameMapper{}
+}
+
+func (pr *noopNameMapper) toEndpointName(txtDNSName string) string {
```

Comment:
You don't modify the underlying struct, so there is no need to use a ptr receiver.
use ```func (pr noopNameMapper) toEndpointName(txtDNSName string) string```